### DISCUSSION TITLE
Uploading Changes to the Cloud

### DISCUSSION THREAD

**Discussion by zzulueta** [microsoft/AI-Chat-App-Hack#54] [Q&A] [answered]
Hi, I made my changes in my local and would like to put them in the cloud in my deployment. I am however unable to do so..

error: failed to push some refs to 'https://github.com/zzulueta/azure-search-openai-demo.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
2024-02-02 03:48:22.462 [info] > git config --get commit.template [2ms]
2024-02-02 03:48:22.502 [info] > git for-each-ref --format=%(refname)%00%(upstream:short)%00%(objectname)%00%(upstream:track)%00%(upstream:remotename)%00%(upstream:remoteref) refs/heads/main refs/remotes/main [3ms]
2024-02-02 03:48:22.551 [info] > git status -z -uall [9ms]
2024-02-02 03:48:27.422 [info] > git ls-tree -l 

I did a pull and got this:
 $ git pull origin main
From https://github.com/zzulueta/azure-search-openai-demo
 * branch            main       -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.

Should I merge or rebase?

**Comment by zzulueta** [comment]
I deleted everything, created a new codespace and made changes. I dont get any issues now committing changes but i dont know why changes arent reflecting in the cloud site

**Reply by pamelafox** [reply, marked as answer]
To make changes to the deployed code, you'll need to run `azd deploy`

**Reply by zzulueta** [reply]
This works thanks!!

**Comment by pallavigadepalli** [comment]
@pamelafox , if I want to continue experimenting with this code post hackathon is it available to us? example the front end piece that was used to demonstrate some of the portions chris Ayers, you, Matt and Khelan used can we continue to look at it and use it later? Thank you!

**Reply by pamelafox** [reply]
All of the code we've shown in our demonstrations are in open-source repositories, so yep, they will remain available.
